### ISSUE TITLE
bug? interaction between multiple pipenv shells

### ISSUE THREAD

**Issue by chuck1** [kennethreitz/pipenv#421]
I'm working on multiple projects at once, all using pipenv. I've noticed that if I start pipenv shell in project A root, and then move to project B and try to start pipenv shell again, it tells me that the shell for B is already activated. So pipenv is automatically switching between shells? Is this the desired behavior?

More concerning is that when I pipenv install an external package in project A, and then move to project B root and pipenv install it again I get something like:

```
$ pipenv shell
Shell for ws-storage-5jC_NMuk already activated. No action taken to avoid nested environments.
$ pipenv install --dev toml
Installing toml...
Requirement already satisfied: toml in /home/bob/.local/share/virtualenvs/ws-sheets-server-rkQt47xA/lib/python3.5/site-packages

Adding toml to Pipfile's [dev-packages]...
```

pipenv says its already installed, but references another project's virtualenv directory. Are my projects sharing their locally installed packages?

**Comment by chuck1** [comment]
Further investigation...
```
$ mkdir A
$ mkdir B
$ cd A
$ pipenv shell
Creating a Pipfile for this project...
Creating a virtualenv for this project...
⠋Using base prefix '/usr'
New python executable in /home/bob/.local/share/virtualenvs/A-TaiTb0IT/bin/python3
Also creating executable in /home/bob/.local/share/virtualenvs/A-TaiTb0IT/bin/python
Installing setuptools, pip, wheel...done.

Virtualenv location: /home/bob/.local/share/virtualenvs/A-TaiTb0IT
Launching subshell in virtual environment. Type 'exit' or 'Ctrl+D' to return.
$ cd ../B
$ echo $PATH
/home/bob/.local/share/virtualenvs/A-TaiTb0IT/bin:/home/bob/bin:/home/bob/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/bob/.rvm/bin:/home/bob/.rvm/bin
$ pipenv shell
Creating a Pipfile for this project...
Shell for B-E3lxBntI already activated. No action taken to avoid nested environments.
$ echo $PATH
/home/bob/.local/share/virtualenvs/A-TaiTb0IT/bin:/home/bob/bin:/home/bob/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/bob/.rvm/bin:/home/bob/.rvm/bin

```
When you switch from project A to project B and run pipenv shell, pipenv says its in the B shell, but PATH tells us otherwise.

Side note, as you might have guessed, my use case involves frequently switching between projects and needing to be in the pipenv shell in each one. Before I realized the above was happening I was getting very confused. Now I at least know whats going on, but it takes an annoyingly long time to switch back and forth and make sure I'm always in the right shell...

**Comment by nateprewitt** [comment]
Hey @chuck1, thanks for taking the time to dig into this.

So first off, that error message was definitely not what we intended. I've added 467c593 to address this by checking the environment variable set by virtualenv rather than assuming the activated shell is for your current project. This will be available in the next release.

Secondly, regarding the workflow you're using. What operations are you performing that require you to be inside the shell? Is it specifically because `pipenv run` doesn't work, or that you're running many commands and trying to avoid using `pipenv run` for each? I don't think we want to try guessing which virtualenv people want to use based on directory for the time being. That means whatever project you run `pipenv shell` in will be the active virtualenv used until you exit the session.

I'm going to close this out as resolved with 467c593, but feel free to answer the above questions or ask any others you may have. Thanks!

**Comment by chuck1** [comment]
THank you for your response, @nateprewitt! I was running alot of commands and trying to avoid using ``pipenv run`` each time. But one simple solution would be to keep multiple terminals open. I agree that once a ``pipenv shell`` for one project is entered, it should remain the active shell until the user exits. To switch shells automatically seems like a bad idea.
